File: lib/src/core/reflection/platform_reflection_capabilities.dart
library angular2.src.core.reflection.platform_reflection_capabilities;

import "package:angular2/src/facade/lang.dart" show Type;
import "types.dart" show GetterFn, SetterFn, MethodFn;

abstract class PlatformReflectionCapabilities {
  bool isReflectionEnabled();
  Function factory(Type type);
  List<dynamic> interfaces(Type type);
  List<List<dynamic>> parameters(dynamic type);
  List<dynamic> annotations(dynamic type);
  Map<String, List<dynamic>> propMetadata(dynamic typeOrFunc);
  GetterFn getter(String name);
  SetterFn setter(String name);
  MethodFn method(String name);
  String importUri(Type type);
}

File: packages-src/core/src/core/reflection/platform_reflection_capabilities.ts
import {Type} from 'angular2/src/facade/lang';
import {GetterFn, SetterFn, MethodFn} from './types';

export interface PlatformReflectionCapabilities {
  isReflectionEnabled(): boolean;
  factory(type: Type): Function;
  interfaces(type: Type): any[];
  parameters(type: any): any[][];
  annotations(type: any): any[];
  propMetadata(typeOrFunc: any): {[key: string]: any[]};
  getter(name: string): GetterFn;
  setter(name: string): SetterFn;
  method(name: string): MethodFn;
  importUri(type: Type): string;
}
